src/java/org/apache/cassandra/schema/KeyspaceMetadata.java

```diff
@@ -62,6 +64,32 @@ public final class KeyspaceMetadata implements SchemaElement
 {
     public static final Serializer serializer = new Serializer();
 
+    /**
+     * Validates the keyspace name for valid characters and correct length.
+     * Throws an exception if it's invalid.
+     *
+     * @param keyspaceName The name of the keyspace to validate
+     * @param ex           The exception class to throw if validation fails
+     */
+    public static <T extends RequestValidationException> void validateKeyspaceName(String keyspaceName, Class<T> ex)
```

Comment:
what is about using a lambda here to create Exception instance instead of reflection? If a signature for one of used exception classes change the current logic will detect it only in runtime, with a lambda it will be found during a compilation.